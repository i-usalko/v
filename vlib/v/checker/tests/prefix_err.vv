b := &true
_ := &10
_ := &"Hi"
_ := &"${b}"

a := 1
_ = *a
a <- 4

_ = ~true
_ = !4
