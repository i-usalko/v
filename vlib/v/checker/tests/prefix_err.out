vlib/v/checker/tests/prefix_err.vv:1:6: error: cannot take the address of a bool
    1 | b := &true
      |      ^
    2 | _ := &10
    3 | _ := &"Hi"
vlib/v/checker/tests/prefix_err.vv:2:6: error: cannot take the address of an int
    1 | b := &true
    2 | _ := &10
      |      ^
    3 | _ := &"Hi"
    4 | _ := &"${b}"
vlib/v/checker/tests/prefix_err.vv:3:6: error: cannot take the address of a string
    1 | b := &true
    2 | _ := &10
    3 | _ := &"Hi"
      |      ^
    4 | _ := &"${b}"
    5 |
vlib/v/checker/tests/prefix_err.vv:4:6: error: cannot take the address of a string
    2 | _ := &10
    3 | _ := &"Hi"
    4 | _ := &"${b}"
      |      ^
    5 |
    6 | a := 1
vlib/v/checker/tests/prefix_err.vv:7:5: error: invalid indirect of `int`
    5 |
    6 | a := 1
    7 | _ = *a
      |     ^
    8 | a <- 4
    9 |
vlib/v/checker/tests/prefix_err.vv:8:1: error: cannot push on non-channel `int`
    6 | a := 1
    7 | _ = *a
    8 | a <- 4
      | ^
    9 |
   10 | _ = ~true
vlib/v/checker/tests/prefix_err.vv:10:5: error: operator ~ only defined on int types
    8 | a <- 4
    9 |
   10 | _ = ~true
      |     ^
   11 | _ = !4
vlib/v/checker/tests/prefix_err.vv:11:5: error: ! operator can only be used with bool types
    9 |
   10 | _ = ~true
   11 | _ = !4
      |     ^
